Extension/src/LanguageServer/settings.ts

```diff
@@ -964,6 +964,9 @@ export class OtherSettings {
     public set filesAssociations(value: any) {
         vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Workspace);
     }
+    public set defaultCompiler(value: any) {
+        vscode.workspace.getConfiguration("files").update("associations", value, vscode.ConfigurationTarget.Global);
```

Comment:
This is setting `files.associations`, and should be updated to set `C_Cpp.default.compilerPath`.  Also please avoid using `any` unless absolutely necessary.  TypeScript is strongly typed, and use of `any` bypasses that strong typing.

Reply:
Whenever I set the type to string, I get a "Type 'String' has no call signatures" error

Reply:
I could help you look into this, if you want.  Generally, we shouldn't use 'any' unless absolutely necessary.